### DISCUSSION TITLE
Error sending DM with media attachment using Twitter API in Java

### DISCUSSION THREAD

**Discussion by MohanKanjula** [redouane59/twittered#445] [Q&A]
Hey , I am trying to reply to a DM with a media attached . So I am using this method in the Library 
`PostDmResponse postDmResponse = twitterClient.createUserDmConversation("participant_id", dmMessage);`

I am building dmMessage object like this

`String templateAttachments= "[{\"media_id\": \"%s\"}]";
   String attachments = String.format(templateAttachments,mediaId);
 DmParameters.DmMessage dmMessage = DmParameters.DmMessage.builder().text("Testing DM with an Image ").attachments(attachments).build() ;
       PostDmResponse postDmResponse = twitterClient.createUserDmConversation("participant_id", dmMessage);`

It is giving me this errror
`[main] ERROR io.github.redouane59.twitter.helpers.AbstractRequestHelper - (POST) Error calling https://api.twitter.com/2/dm_conversations/with/864922074502049793/messages {"errors":[{"parameters":{"$.attachments":["\"[{\\\"media_id\\\":\\\"1647942759276326912\\\"}]\""]},"message":"$.attachments: string found, array expected"}],"title":"Invalid Request","detail":"One or more parameters to your request was invalid.","type":"https://api.twitter.com/2/problems/invalid-request"} - 400
`

As I checked the official twitter API documentation , there attachments is an array of json objects 
`{"text": "Sending a DM with media!", "attachments": [{"media_id": "1455952740635586573"}]`

So in which format should we send the attachment String here ? can you help me with this?

**Comment by AlanNun08** [comment]
I am using java and I am having the same issues with twittered

**Reply by AlanNun08** [reply]
however the issues relies with posting a tweet with a media object
